core/src/main/java/io/aiven/klaw/validation/TopicRequestValidatorImpl.java

```diff
@@ -1,21 +1,244 @@
 package io.aiven.klaw.validation;
 
+import io.aiven.klaw.dao.Topic;
+import io.aiven.klaw.model.ApiResultStatus;
+import io.aiven.klaw.model.PermissionType;
 import io.aiven.klaw.model.TopicRequestModel;
+import io.aiven.klaw.model.TopicRequestTypes;
+import io.aiven.klaw.service.CommonUtilsService;
+import io.aiven.klaw.service.MailUtils;
+import io.aiven.klaw.service.TopicControllerService;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
 import javax.validation.ConstraintValidator;
 import javax.validation.ConstraintValidatorContext;
 import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
 
 @Slf4j
 public class TopicRequestValidatorImpl
     implements ConstraintValidator<TopicRequestValidator, TopicRequestModel> {
 
+  @Autowired private CommonUtilsService commonUtilsService;
+  @Autowired private MailUtils mailService;
+  @Autowired private TopicControllerService topicControllerService;
+
+  private PermissionType permissionType;
+
   @Override
-  public void initialize(TopicRequestValidator constraintAnnotation) {}
+  public void initialize(TopicRequestValidator constraintAnnotation) {
+    this.permissionType = constraintAnnotation.getPermissionType();
+  }
 
   @Override
   public boolean isValid(
       TopicRequestModel topicRequestModel, ConstraintValidatorContext constraintValidatorContext) {
+
+    // Verify if user has access to request for topics
+    if (commonUtilsService.isNotAuthorizedUser(
+        topicControllerService.getPrincipal(), this.permissionType)) {
+      updateConstraint(constraintValidatorContext, ApiResultStatus.NOT_AUTHORIZED.value);
+      return false;
+    }
+
+    // tenant filtering
+    if (!topicControllerService
+        .getEnvsFromUserId(topicControllerService.getUserName())
+        .contains(topicRequestModel.getEnvironment())) {
+      updateConstraint(
+          constraintValidatorContext,
+          "Failure. Not authorized to request topic for this environment.");
+      return false;
+    }
+
+    // check for null topic name
+    if (topicRequestModel.getTopicname() == null) {
+      updateConstraint(constraintValidatorContext, "Failure. Please fill in topic name.");
+      return false;
+    }
+
+    // check for empty topic name
+    if (topicRequestModel.getTopicname().length() == 0) {
```

Comment:
here is a bit tricky thing... 
Do we allow names full of spaces? e.g. one ` ` or 2 spaces `  `?
Or should we check if it is blank?
like
```java
    // check for empty topic name
    if (topicRequestModel.getTopicname().isBlank()) {
      updateConstraint(constraintValidatorContext, "Failure. Please fill in a valid topic name.");
      return false;
    }
```

Reply:
Agree, isBlank is better and for topic names, spaces are not allowed. Updated.